### PR TITLE
Migrate Product Price to `block.json`

### PR DESCRIPTION
**Pull request by kmanijak**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

We want to migrate all of the blocks to `block.json` for consistency and to make sure all of them are correctly parsed by a script that creates Block References doc.

This PR migrates Product Price to `block.json`.

Part of https://github.com/woocommerce/woocommerce/issues/46917.

> [!NOTE]  
> `plugins/woocommerce-blocks/docs/block-references/block-references.md` is auto-generated hence contains more changes from other blocks.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Product Image smoke testing

1. Create new post
2. Add All Products block
3. Enter edit mode and make some edits in Product Price block inspector controls
4. Add Product Collection block
5. Choose any collection
6. Make some edits in Product Price block inspector controls
7. Save and go to frontend
8. Make sure the block is properly rendered, inspector controls settings are reflected

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/price/block.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		},
+		"isDescendentOfSingleProductBlock": {
+			"type": "boolean",
+			"default": false
+		}
+	},
+	"usesContext": [ "query", "queryId", "postId" ],
+	"keywords": [ "WooCommerce" ],
+	"textdomain": "woocommerce",
+	"apiVersion": 3,
```
Comment (ObliviousHarmony) [inline plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/price/block.json:23]: I'm not sure if it matters but _technically_ we're missing some of the configuration present [in the shared config](https://github.com/woocommerce/woocommerce/blob/b62bc4851ac006a764400b2cda2ebf7fbd140076/plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/shared/config.tsx#L29-L38).

```suggestion
	"apiVersion": 3,
	"supports": {
		"html": false
	},
	"ancestor": [
		'woocommerce/all-products',
		'woocommerce/single-product',
		'woocommerce/product-template',
		'core/post-template'
	],
```

**Review by ObliviousHarmony — COMMENTED** [review]
Thanks @kmanijak,

Migrating stuff to new config formats sure is scary 😅

I followed your testing instructions and it _seems_ like the block works just fine. I left one comment but as far as I can tell it actually doesn't seem to break anything not being in `block.json`.

Realistically speaking, we should probably move away from the shared config entirely but that's a different problem.

File: plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/price/block.json (hunk 1 above)
Comment (kmanijak) [reply plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/price/block.json:23]: > Realistically speaking, we should probably move away from the shared config entirely but that's a different problem.

I figured the same. I even started doing so in https://github.com/woocommerce/woocommerce/pull/50920 (it's just a random draft). But shared config is confusing and is a source of a lot of duplication (paradoxically 😅). So basically my idea is:
1. Migrate all of the Product Elements to `block.json`
2. Once we're there, remove shared config and cleanup block definitions.

> I'm not sure if it matters but technically we're missing some of the configuration present [in the shared config](https://github.com/woocommerce/woocommerce/blob/b62bc4851ac006a764400b2cda2ebf7fbd140076/plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/shared/config.tsx#L29-L38).

Re this, good point, I can already start moving the fields over to `block.json` and remove the dependency on shared config 👍

Comment (kmanijak) [reply plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/price/block.json:23]: After second thoughts I decided not to do it. At the moment the `block.json` properties are anyway overwritten by shared config and `supports` in `index.tsx` file [here](https://github.com/woocommerce/woocommerce/pull/50905/files#diff-fd43d1e70b547324974885be9c533448cf5f7013d3c5cea1f60a7580c96243d7R16-R20).

Hence I think it'll be better to do the cleanup once all of the blocks are migrated and having less overwrites is easier to analyse and handle IMO. I'm planning to finalise the migration and do the cleanup in next cooldown.

Comment (ObliviousHarmony) [reply plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/price/block.json:23]: Sorry @kmanijak, I wasn't trying to nerd snipe you 😅 I was thinking that it's something we should do later. The best approach is to do the migration of our blocks and _then_ remove the shared config. This makes sure everything works and then when we pull the shared config we are in a state where we know it _should_ work.

With that in mind it probably makes sense to ensure that _all_ of the shared config is duplicated in all of the migrated JSON files. That will make sure we are able to cleanly remove it later. Assuming we do that we should also leave a comment on the shared config indicating it is deprecated and why.

Comment (kmanijak) [reply plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/price/block.json:23]: Makes sense! Added in https://github.com/woocommerce/woocommerce/pull/50905/commits/5215dfb89546e3d12501214cade2341f8c4d5202

**Review by ObliviousHarmony — APPROVED** [review]
Awesome work @kmanijak, thanks a lot!

**Pull request merged**